src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,237 @@
+    *ppLogging = pProfilerManagerLogging.Detach();
+    return S_OK;
+}
+
+HRESULT CProfilerManagerWrapper::SetLoggingHost(_In_opt_ IProfilerManagerLoggingHost* pLoggingHost)
+{
+    return m_pProfilerManager->SetLoggingHost(pLoggingHost);
+}
+
+HRESULT CProfilerManagerWrapper::GetAppDomainCollection(_Out_ IAppDomainCollection** ppAppDomainCollection)
+{
+    return m_pProfilerManager->GetAppDomainCollection(ppAppDomainCollection);
+}
+
+
+HRESULT CProfilerManagerWrapper::CreateSignatureBuilder(_Out_ ISignatureBuilder** ppSignatureBuilder)
+{
+    return m_pProfilerManager->CreateSignatureBuilder(ppSignatureBuilder);
+}
+
+HRESULT CProfilerManagerWrapper::GetInstrumentationMethod(_In_ REFGUID cslid, _Out_ IUnknown** ppUnknown)
+{
+    return m_pProfilerManager->GetInstrumentationMethod(cslid, ppUnknown);
+}
+
+HRESULT CProfilerManagerWrapper::RemoveInstrumentationMethod(_In_ IInstrumentationMethod* pInstrumentationMethod)
+{
+    return m_pProfilerManager->RemoveInstrumentationMethod(pInstrumentationMethod);
+}
+
+HRESULT CProfilerManagerWrapper::AddInstrumentationMethod(
+    _In_ BSTR bstrModulePath,
+    _In_ BSTR bstrName,
+    _In_ BSTR bstrDescription,
+    _In_ BSTR bstrModule,
+    _In_ BSTR bstrClassGuid,
+    _In_ DWORD dwPriority,
+    _Out_ IInstrumentationMethod** ppInstrumentationMethod)
+{
+    return m_pProfilerManager->AddInstrumentationMethod(
+        bstrModulePath,
+        bstrName,
+        bstrDescription,
+        bstrModule,
+        bstrClassGuid,
+        dwPriority,
+        ppInstrumentationMethod);
+}
+
+// IProfilerManager2 Methods
+HRESULT CProfilerManagerWrapper::DisableProfiling()
+{
+    return m_pProfilerManager->DisableProfiling();
+}
+
+HRESULT CProfilerManagerWrapper::ApplyMetadata(_In_ IModuleInfo* pMethodInfo)
+{
+    return m_pProfilerManager->ApplyMetadata(pMethodInfo);
+}
+
+// IProfilerManager3 Methods
+HRESULT CProfilerManagerWrapper::GetApiVersion(_Out_ DWORD* pApiVer)
+{
+    return m_pProfilerManager->GetApiVersion(pApiVer);
+}
+
+// IProfilerManager4 Methods
+HRESULT CProfilerManagerWrapper::GetGlobalLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    return m_pProfilerManager->GetGlobalLoggingInstance(ppLogging);
+}
+
+
+// IProfilerManager5 Methods
+HRESULT CProfilerManagerWrapper::IsInstrumentationMethodRegistered(_In_ REFGUID clsid, _Out_ BOOL* pfRegistered)
+{
+    return m_pProfilerManager->IsInstrumentationMethodRegistered(clsid, pfRegistered);
+}
+
+// IProfilerManagerLogging Methods
+/*
+ * Inject Instrumentation Method ClassId
+ */
+HRESULT CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
+{
+    HRESULT hr = S_OK;
+    WCHAR wszBuffer[LogEntryMaxSize] = { 0 };
+    tstring tsMessage = wszMessage;
+    tstring tsEscapedMessage;
+    for (tstring::iterator it = tsMessage.begin(); it < tsMessage.end(); it++)
+    {
+        // Escape any existing format strings to avoid conflicts with the FORMATTABLE_PREFIX
+        if (*it == L'%')
```

Comment:
Maybe add a better comment as to what is happening here. If I understand correctly, you are duplicating the % characters since %% is how you escape a single % character in a format string that you don't want treated as a format specifier.

Reply:
Comment is in refactored method "EscapeFormatSpecifiers"

---
In reply to: [341216601](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r341216601) [](ancestors = 341216601)